src/main/java/org/reactome/orthoinference/EventsInferrer.java

```diff
@@ -334,4 +345,33 @@ private static void setInstanceEdits(int personId) throws Exception
 		EWASInferrer.setInstanceEdit(instanceEditInst);
 		PathwaysInferrer.setInstanceEdit(instanceEditInst);
 	}
+
+	/**
+	 * Download the Wormbase file that contains the WBGene IDs mapped to gene names, and then create a mapping of IDs to names.
+	 * @param wormbaseUrl -- URL to Wormbase file to be downloaded and processed.
+	 * @return -- Map<String, List,<String>> of WBGene IDs to gene names.
+	 * @throws IOException
+	 */
+	private static Map<String, List<String>> downloadAndProcessWormbaseFile(URL wormbaseUrl) throws IOException {
+		File wormbaseFile = Paths.get(wormbaseUrl.toString()).getFileName().toFile();
+		Map<String, List<String>> wormbaseMappings = new HashMap<>();
+		if (!wormbaseFile.exists()) {
+			FileUtils.copyURLToFile(wormbaseUrl, wormbaseFile);
+		}
+		BufferedReader br = new BufferedReader(new InputStreamReader(new GZIPInputStream(new FileInputStream(wormbaseFile))));
+		String line = null;
+		while ((line = br.readLine()) != null) {
+			String wormbaseGeneId = line.split(",")[1];
+			String wormbaseGeneName = line.split(",")[2];
+			if (!wormbaseGeneId.isEmpty() && !wormbaseGeneName.isEmpty()) {
+				if (wormbaseMappings.get(wormbaseGeneId) != null) {
+					wormbaseMappings.get(wormbaseGeneId).add(wormbaseGeneName);
+				} else {
+					wormbaseMappings.put(wormbaseGeneId, new ArrayList<>(Arrays.asList(wormbaseGeneName)));
```

Comment:
This could be done as `wormbaseMappings.computeIfAbsent(wormbaseGeneId, k -> new ArrayList<>()).add(wormbaseGeneName)`

Reply:
Done